amt/core/exception_handlers.py

```diff
@@ -5,23 +5,43 @@
 from fastapi.responses import HTMLResponse
 from starlette.exceptions import HTTPException as StarletteHTTPException
 
+from amt.api.deps import templates
+
 logger = logging.getLogger(__name__)
 
 
 async def http_exception_handler(request: Request, exc: StarletteHTTPException) -> HTMLResponse:
     logger.debug(f"http_exception_handler: {exc.status_code} - {exc.detail}")
-    return HTMLResponse(
+
+    if request.state.htmx:
+        return templates.TemplateResponse(
+            request,
+            "errors/_HTMXHTTPException.html.j2",
+            {"status_code": exc.status_code, "status_message": exc.detail},
+            status_code=exc.status_code,
+        )
+
+    return templates.TemplateResponse(
+        request,
+        "errors/HTTPException.html.j2",
+        {"status_code": exc.status_code, "status_message": exc.detail},
         status_code=exc.status_code,
-        content=f"<p>{exc.detail}</p>",
     )
 
 
-async def validation_exception_handler(_request: Request, exc: RequestValidationError) -> HTMLResponse:
+async def validation_exception_handler(request: Request, exc: RequestValidationError) -> HTMLResponse:
     logger.debug(f"validation_exception_handler: {exc.errors()}")
     errors = exc.errors()
     messages: list[str] = [f"{error['loc'][-1]}: {error['msg']}" for error in errors]
 
-    return HTMLResponse(
-        status_code=status.HTTP_400_BAD_REQUEST,
-        content=f"<h1>Invalid Request</h1><p>{messages}</p>",
+    if request.state.htmx:
```

Comment:
There is no test that hits this condition.

Reply:
solved